crates/uitest/fixtures/ty/trait_bound/hkt_bound.snap

```diff
@@ -0,0 +1,12 @@
+---
+source: crates/uitest/tests/ty.rs
+expression: diags
+input_file: crates/uitest/fixtures/ty/trait_bound/hkt_bound.fe
+---
+error[6-0002]: given trait argument kind mismatch
+  ┌─ hkt_bound.fe:1:17
+  │
+1 │ pub trait Trait<T: HktTrait> {}
+  │                 ^ expected `(* -> *)` kind, but `T` has `*` kind
```

Comment:
This error is confusing to me. Is this because any generic type `T` has kind `*` by default, unless explicitly specified? Changing this to `T: HktTrait + * -> *` eliminates the error. However, `T: HktTrait` should imply that `T: * -> *`. Is this too complicated to deduce? (Not a major issue of course, I'm mostly curious)